### PR TITLE
Playlist management

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Features implemented 
- Users can create playlists from the video listing page as well as the playlist management page. 
- Delete a playlist.
- Add videos to the playlist.
- Remove videos from the playlist.

### CODE REVIEW

File: src/components/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,41 @@
+.modal-background {
+  z-index: 1500;
```
Comment (Chakravarthi0) [inline src/components/modal/modal.css:2]: I would suggest to use z-index values incrementally. You might face issues if you try to add z-index in some other component.

File: src/components/playlist-card/PlaylistCard.jsx [hunk 2]
```diff
@@ -1,21 +1,51 @@
...
+import { thumbnailLink } from "../../utils";
 import "./playlistCard.css";
 
-const PlaylistCard = () => {
+const PlaylistCard = ({ playlist }) => {
+  const { _id, title, videos } = playlist;
+
+  const { deletePlaylist } = usePlaylist();
+
+  const [showOptions, setShowOptions] = useState(false);
```
Comment (Chakravarthi0) [inline src/components/playlist-card/PlaylistCard.jsx]: You can move it to the top.

File: src/components/playlist-card/PlaylistCard.jsx [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
   return (
     <div className="playlist-card">
-      <div className="playlist-card-img-container">
+      <Link to={`/playlist/${playlist._id}`} className="playlist-card-img-container">
         <img
           className="responsive-img"
-          src="https://i.ytimg.com/vi/_erVOAbz420/hqdefault.jpg?sqp=-oaymwEcCOADEI4CSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCLTS3HE2hVrjsA_n8UH2pSH7XXrw"
-          alt="thumbnail-img"
+          src={
+            videos.length ? thumbnailLink(videos[0]?._id) : "/assets/undraw_going_up_re_86kg.svg"
```
Comment (Chakravarthi0) [inline src/components/playlist-card/PlaylistCard.jsx]: videos?.length

File: src/components/sidebar/Sidebar.jsx [hunk 4]
```diff
@@ -1,45 +1,57 @@
-import { Link } from "react-router-dom";
 import "./sidebar.css";
+import { NavLink } from "react-router-dom";
 
 const Sidebar = () => {
+  const activeClass = isActive => {
+    console.log(isActive);
+  };
```
Comment (Chakravarthi0) [inline src/components/sidebar/Sidebar.jsx]: You can remove this, if not needed.

File: src/context/playlistContext.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+        const res = await addToPlaylistService(token, playlistId, video);
+
+        if (res.status === 201) {
+          playlistDispatch({ type: UPDATE_PLAYLIST, payload: res.data.playlist });
+          toast.success("Video added to playlist");
+        }
+      } catch (err) {
+        playlistDispatch({ type: SET_ERROR, payload: err.response.data.errors });
+        toast.error(err.response.data.errors);
```
Comment (Chakravarthi0) [inline src/context/playlistContext.js]: You can use: err.response.data.error[0]

File: src/pages/explore/Explore.jsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { Link } from "react-router-dom";
+import { useState } from "react";
```
Comment (Chakravarthi0) [inline src/pages/explore/Explore.jsx]: I think, you can remove it

File: src/pages/playlist-listing/PlaylistListing.jsx [hunk 7]
```diff
@@ -1,16 +1,76 @@
...
-      <PlaylistCard />
-      <PlaylistCard />
-      <PlaylistCard />
-      <PlaylistCard />
-      <PlaylistCard />
-      <PlaylistCard />
-    </main>
+    <div>
+      {/* {loading && <Loading />}
+      {error && <span>{error}</span>} */}
```
Comment (Chakravarthi0) [inline src/pages/playlist-listing/PlaylistListing.jsx]: You can remove this

File: src/pages/playlist-listing/PlaylistListing.jsx [hunk 8]
```diff
@@ -1,16 +1,76 @@
...
+                    className="btn btn-primary"
+                    onClick={e => {
+                      e.preventDefault();
+                      if (playlistName.length) {
+                        createPlaylist(playlistName);
+                        setPlaylistName("");
+                        setIsCreating(false);
+                      } else {
+                        toast.error("Playlist name can not be empty");
+                      }
```
Comment (Chakravarthi0) [inline src/pages/playlist-listing/PlaylistListing.jsx]: You can extract it into a function

**Review by Chakravarthi0 — APPROVED** [review]
UI looks good.

**Review by BelideNikhil — APPROVED** [review]
The code is clean and is UI is good and works as expected
Apart from suggested changes, rest of the code is nicely written,

File: src/components/modal/modal.css (hunk 1 above)
Comment (msdeshmukh009) [reply src/components/modal/modal.css:2]: Ok, I will find a way around it.

File: src/components/playlist-card/PlaylistCard.jsx (hunk 2 above)
Comment (msdeshmukh009) [reply src/components/playlist-card/PlaylistCard.jsx]: done

File: src/components/playlist-card/PlaylistCard.jsx (hunk 3 above)
Comment (msdeshmukh009) [reply src/components/playlist-card/PlaylistCard.jsx]: done

File: src/components/sidebar/Sidebar.jsx (hunk 4 above)
Comment (msdeshmukh009) [reply src/components/sidebar/Sidebar.jsx]: done

File: src/context/playlistContext.js (hunk 5 above)
Comment (msdeshmukh009) [reply src/context/playlistContext.js]: done

File: src/pages/explore/Explore.jsx (hunk 6 above)
Comment (msdeshmukh009) [reply src/pages/explore/Explore.jsx]: done

File: src/pages/playlist-listing/PlaylistListing.jsx (hunk 7 above)
Comment (msdeshmukh009) [reply src/pages/playlist-listing/PlaylistListing.jsx]: done

File: src/pages/playlist-listing/PlaylistListing.jsx (hunk 8 above)
Comment (msdeshmukh009) [reply src/pages/playlist-listing/PlaylistListing.jsx]: Done. Thanks for the review.

**Pull request merged**